File: ObgApp/src/app/models/owner.ts
import { Pharmacy } from "./pharmacy";
import { RoleUser, User } from "./user";

export class Owner extends User {

    pharmacy: Pharmacy | null;

    constructor(name: string | null, code: number | null, email: string | null, password: string | null, address: string | null, role: RoleUser | null, registerDate: string | null, pharmacy: Pharmacy | null) {
        super(name, code, email, password, address, role, registerDate);
        this.pharmacy = pharmacy;
    }
}

File: Backend/obg.Domain/Entities/Owner.cs
﻿using obg.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace obg.Domain.Entities
{
    public class Owner : User
    {
        public Pharmacy Pharmacy { get; set; }
        
        public Owner() { }
        public Owner(string name, int code, string email, string password, string address, RoleUser role, string registerDate, Pharmacy pharmacy) : base(name, code, email, password, address, role, registerDate)
        {
            Pharmacy = pharmacy;
        }

    }
}
